challenge_serving_drinks/src/challenge_serving_drinks/serve_one_drink.py

```diff
@@ -0,0 +1,162 @@
+# ROS
+import smach
+
+# Robot smach states
+import robot_smach_states as states
+import robot_smach_states.util.designators as ds
+
+from robot_skills.util.entity import Entity
+from robocup_knowledge import knowledge_loader
+from robot_skills.robot import Robot
+
+# Serving drinks
+from .sd_states import DescriptionStrDesignator
+from .get_order import GetOrder
+
+# Knowledge
+CHALLENGE_KNOWLEDGE = knowledge_loader.load_knowledge("challenge_serving_drinks")
+
+
+class ServeOneDrink(smach.StateMachine):
+    """ Serves on drink to an operator
+
+    """
+    def __init__(self, robot, bar_designator, room_designator, objects_list_des, unav_drink_des):
+        # type: (Robot, int) -> str
+        """ Initialization method
+
+        :param robot: robot api object
+        :param idx: index of the drink that is served (used for the operator name)
+        """
+        smach.StateMachine.__init__(self, outcomes=["succeeded", "failed", "aborted"])
+
+        # Designators
+        arm_designator = ds.UnoccupiedArmDesignator(robot=robot, arm_properties={}, name='arm_des')
+
+        drink_str_designator = ds.VariableDesignator(resolve_type=str, name='drink_str_des')
+        drink_designator = ds.EdEntityDesignator(robot=robot, type_designator=drink_str_designator, name='drink_des')
+
+        operator_name = ds.VariableDesignator(resolve_type=str, name='name_des')
+        operator_designator = ds.VariableDesignator(resolve_type=Entity, name='operator_des')
+
+        with self:
+
+            # Get order
+            smach.StateMachine.add(
+                "GET_ORDER",
+                GetOrder(
```

Comment:
Also style fixes in this file.

Reply:
Will be reflected in the new PR (styling)